### PR TITLE
Fix broken intradoc links and test rustdoc in CI

### PR DESCRIPTION
**Pull request by MarijnS95**
Let us make sure our public crates have sane docs over at https://docs.rs/gpu-allocator/ :)

### CODE REVIEW

File: src/vulkan/mod.rs [hunk 1]
```diff
@@ -60,19 +60,19 @@ impl Allocation {
...
     }
 
-    /// Returns the `vk::DeviceMemory` object that is backing this allocation.
+    /// Returns the [`vk::DeviceMemory`] object that is backing this allocation.
     /// This memory object can be shared with multiple other allocations and shouldn't be freed (or allocated from)
     /// without this library, because that will lead to undefined behavior.
     ///
     /// # Safety
-    /// The result of this function can safely be used to pass into `bind_buffer_memory` (`vkBindBufferMemory`),
-    /// `bind_texture_memory` (`vkBindTextureMemory`) etc. It's exposed for this reason. Keep in mind to also
```
Comment (MarijnS95) [inline src/vulkan/mod.rs:69]: @max-traverse I suppose you meant `bind_image_memory` here? Also, there's a `2` variant of both functions, but I don't think we should bother to list them here (unless wanting to write a nice summation).

Comment (MarijnS95) [reply src/vulkan/mod.rs:69]: Also, `# Safety` should clarify what safety constraints the caller should be held accountable for which is not explained in this block.

Comment (max-traverse) [reply src/vulkan/mod.rs:69]: oh `bind_texture_memory` should 100% be `bind_image_memory`. Not sure how this ended up in the safety section.

**Review by max-traverse — APPROVED** [review]
Nice clean-up! Just need to figure out what to do with those bind comments in the safety section.

**Comment by MarijnS95** [comment]
> Just need to figure out what to do with those bind comments in the safety section.

Let's merge this, I'll let you take care of some doc rewriting in a new PR ;)

**Pull request merged**